### PR TITLE
fix: getPipelines() returns all pipelines instead of defaulting to maven

### PR DESCRIPTION
**Pull request by bartoszmajsak**
`appLauncherPipelineService#getPipelines()` had default filter set to `maven` which made it impossible to fetch all pipelines exposed by Launcher Backend API. 

Part of https://github.com/openshiftio/openshift.io/issues/4140

### CODE REVIEW

File: src/app/space/app-launcher/services/app-launcher-pipeline.service.ts [hunk 1]
```diff
@@ -37,15 +37,15 @@ export class AppLauncherPipelineService implements PipelineService {
     }));
   }
 
-  getPipelines(filterByRuntime: string = 'maven'): Observable<Pipeline[]> {
+  getPipelines(filterByRuntime?: string): Observable<Pipeline[]> {
```
[the remark this answers is not in the document]
Comment (bartoszmajsak) [reply src/app/space/app-launcher/services/app-launcher-pipeline.service.ts]: There is only [one caller ](https://github.com/fabric8-launcher/ngx-launcher/blob/ea5ee6ac3a6e8fa4d8802e083d147dd80b66d228/src/app/launcher/create-app/release-strategy-createapp-step/release-strategy-createapp-step.component.ts#L38-L45) in `ngx-launcher`, but it defaults to Maven regardless. Also notice that the interface signature makes this [parameter optional](https://github.com/fabric8-launcher/ngx-launcher/blob/dea26e40fe9fff572ac69334d158b0fee401c430/src/app/launcher/service/pipeline.service.ts#L13)

File: src/app/space/app-launcher/services/app-launcher-pipeline.service.ts [hunk 2]
```diff
@@ -37,15 +37,15 @@ export class AppLauncherPipelineService implements PipelineService {
...
     let runtimeEndPoint: string = this.END_POINT + this.API_BASE;
     return this.options.flatMap((option) => {
       return this.http.get(runtimeEndPoint, option)
         .map(response => response.json() as Pipeline[])
         .map(pipelines => {
           // needs to filter out associated pipelines from list of pipelines
           return pipelines.filter(item => {
-            return item.platform === filterByRuntime;
+            return item.platform === (filterByRuntime || item.platform);
```
[the remark this answers is not in the document]
Comment (bartoszmajsak) [reply src/app/space/app-launcher/services/app-launcher-pipeline.service.ts]: Thanks for `({}) =>`  though. It's neat.

File: src/app/space/app-launcher/services/app-launcher-pipeline.service.spec.ts [hunk 3]
```diff
@@ -1,86 +1,129 @@
...
-    Pipeline,
-    TokenProvider
-} from 'ngx-launcher';
-
-import { FABRIC8_FORGE_API_URL } from '../../../shared/runtime-console/fabric8-ui-forge-api';
-import { NewForgeConfig } from '../shared/new-forge.config';
-import { AppLauncherPipelineService } from './app-launcher-pipeline.service';
+import {TestBed} from '@angular/core/testing';
+import {HttpModule, Response, ResponseOptions, XHRBackend} from '@angular/http';
+import {MockBackend} from '@angular/http/testing';
```
Comment (invincibleJai) [inline src/app/space/app-launcher/services/app-launcher-pipeline.service.spec.ts]: my tslint is complaining for space i.e `{ MockBackend }`

File: src/app/space/app-launcher/services/app-launcher-pipeline.service.spec.ts [hunk 4]
```diff
@@ -1,86 +1,129 @@
...
+  it('should return empty pipelines collection when no match found for the platform name', (done: DoneFn) => {
+    mockBackend.connections.subscribe((connection) => {
+      connection.mockRespond(new Response(new ResponseOptions({
+        body: JSON.stringify(mockPipelines())
+      })));
+    });
+
+    appLauncherPipelineService.getPipelines('rust').subscribe(response => {
+      let pipelines: Pipeline[] = response;
+      expect(pipelines.length).toBe(0)
```
Comment (invincibleJai) [inline src/app/space/app-launcher/services/app-launcher-pipeline.service.spec.ts]: missing `;`

File: src/app/space/app-launcher/services/app-launcher-pipeline.service.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (christianvogt) [reply src/app/space/app-launcher/services/app-launcher-pipeline.service.ts]: @joshuawilson @bartoszmajsak yes double pipe is very common and am fine with double pipes in general. I only thought that it was odd to compare a value to itself in order to negate the effects of the filter.
Leave as is.

**Comment by jiekang** [comment]
The last jenkins build failed with
```
ERROR in [at-loader] ./src/app/space/app-launcher/services/app-launcher-pipeline.service.spec.ts:50:7 

    TS2352: Type '({ id: string; platform: string; name: string; description: string; stages: [{ name: string; desc...' cannot be converted to type 'Pipeline[]'.

  Type '{ id: string; platform: string; name: string; description: string; stages: [{ name: string; descr...' is not comparable to type 'Pipeline'.

    Type '{ id: string; platform: string; name: string; description: string; stages: [{ name: string; descr...' is not comparable to type 'Pipeline'.

      Object literal may only specify known properties, and 'description' does not exist in type 'Pipeline'.
```

**Comment by joshuawilson** [comment]
@bartoszmajsak our IDEs should show lint issues and running anything from webpack locally should throw lint errors.

**Comment by bartoszmajsak** [comment]
@joshuawilson this approach is reactive. I was more thinking about some proactive/automated way.

**Comment by jiekang** [comment]
I've had that happen before but I've never investigated. At a glance the centos and jenkins builds seem to be doing the same thing, but the centos one doesn't catch these kind of errors.

**Comment by bartoszmajsak** [comment]
@jiekang thanks for spotting it. Fix is coming "shortly" :beers:

**Comment by bartoszmajsak** [comment]
[test]

File: src/app/space/app-launcher/services/app-launcher-pipeline.service.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (bartoszmajsak) [reply src/app/space/app-launcher/services/app-launcher-pipeline.service.ts]: Spot on @christianvogt. I was only looking at "create app" flow, whereas we also do call it in the "import flow". 

It revealed a bigger issue though https://github.com/openshiftio/openshift.io/issues/4154. For the time being let's bring back this behaviour until we have proper import flow in place.

**Pull request merged**